// shim-loader/lib/transform/determineShimOptions.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = determineShimOptions;

var _isArray = require('lodash/isArray');

var _isArray2 = _interopRequireDefault(_isArray);

var _isString = require('lodash/isString');

var _isString2 = _interopRequireDefault(_isString);

var _precond = require('precond');

var _precond2 = _interopRequireDefault(_precond);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function determineShimOptions(moduleFile, shimConfig) {
  _precond2.default.checkIsString(moduleFile, 'Module name/file must be an string');
  _precond2.default.checkIsObject(shimConfig, 'Shimconfig for module \'' + moduleFile + '\' must be an object');

  var exported = shimConfig.exports,
      deps = shimConfig.deps,
      commonjs = shimConfig.commonjs,
      amd = shimConfig.amd;


  if (typeof exported !== 'undefined') {
    _precond2.default.checkArgument((0, _isArray2.default)(exported) || (0, _isString2.default)(exported), 'exports for module \'' + moduleFile + '\' must be an string or array of strings');
  }

  if (typeof deps !== 'undefined') {
    _precond2.default.checkIsArray(deps, 'deps for module \'' + moduleFile + '\' must be an array of strings');
  } else {
    deps = [];
  }

  if (typeof commonjs !== 'undefined') {
    _precond2.default.checkIsBoolean(commonjs, 'commonjs for module \'' + moduleFile + '\' must be an boolean');
  }

  if (typeof amd !== 'undefined') {
    _precond2.default.checkIsBoolean(amd, 'amd for module \'' + moduleFile + '\' must be an boolean');
  }

  _precond2.default.checkState(!(typeof exported !== 'undefined' && amd), 'exports option for module \'' + moduleFile + '\' is not compatible with activated amd flag. Make sure that you either use global or AMD module style.');
  _precond2.default.checkState(!(typeof exported !== 'undefined' && commonjs), 'exports option \'' + moduleFile + '\' is not compatible with activated commonjs flag. Make sure that you either use global or CommonJS module style.');

  var module = !exported && !deps.length && typeof commonjs === 'undefined' || !!commonjs;
  var require = module;
  var define = !exported && !deps.length && typeof amd === 'undefined' || amd;

  return {
    deps: deps,
    exported: exported,
    context: 'global',
    module: module,
    require: require,
    define: define
  };
}
